tests/merge.test.ts

```diff
@@ -0,0 +1,100 @@
+import { EventPayload } from '../src/EventPayload';
+import { merge } from '../src/merge';
+
+describe('Merge Function', () => {
+    it('should merge example 1 correctly', () => {
```

Comment:
We shouldn't have the description reference the spec. Let's rename them to something like:
- "merging primitive values"
- "merging objects (no overlap)"
- "merging objects (with overlap)"
If you think of a better descriptor, go for it. These are just possible ideas.